### ISSUE TITLE
Apply rule RequiredStatusChange (1025) for both Requests and Responses

### ISSUE THREAD

**Issue by guillaume-pansier** [criteo/openapi-comparator#26]
Hello,

I would like to have the RequiredStatusChange fail for requests and responses, as for now there is no errors for the following contract change:
```
"MyEntityName": {
      "required": [
        "myProperty1",
-       "myProperty2"
      ],
```

I checked the implementation, and it bypasses the check for responses:
```
 if (oldParameter.IsRequired() == newParameter.IsRequired() || context.Direction == DataDirection.Response)
                return;
```

The description of the rule, "Checks whether an existing property's required status is changed from the previous specification.", doesn't mention anything about this bypass. Furthermore, I believe changing the "required" status in a response is a breaking change as well.

I'll see if I can investigate and push a PR later on.

**Comment by bubianchi-criteo** [comment]
Thank you for reporting.
The code you mention is found in [ParameterComparator](https://github.com/criteo/openapi-comparator/blob/c9dc7a52655bafc96d1f593815d539e781d0b03c/src/Criteo.OpenApi.Comparator/Comparators/ParameterComparator.cs#L100) and concerns an operation parameter either in the path, query, header or cookie.
I think you are more concerned about required properties in schema here, aren't you?

**Comment by bubianchi-criteo** [comment]
The `ResponseComparator` uses eventually the `SchemaComparator` (see [OpenApiDocumentComparator constructor](https://github.com/criteo/openapi-comparator/blob/c9dc7a52655bafc96d1f593815d539e781d0b03c/src/Criteo.OpenApi.Comparator/Comparators/OpenApiDocumentComparator.cs#L24) that [checks](https://github.com/criteo/openapi-comparator/blob/c9dc7a52655bafc96d1f593815d539e781d0b03c/src/Criteo.OpenApi.Comparator/Comparators/SchemaComparator.cs#L546) only if there is no new required property in the schema.

**Comment by guillaume-pansier** [comment]
You're right, the problem I had was with the schema, it wasn't about parameters.

If I understand correctly what you're suggesting, there should be a new rule added to [list of already existing rules](https://github.com/criteo/openapi-comparator/tree/main/documentation) right ?

I tried with a fork of the repository, so that we can discuss on something concrete.
The behavior seems OK to me, except for one thing: if a required property get removed entirely, the tool will now report 2 warnings: one about the "removed property", one about the "removed required". It seems a bit redundant to me, but maybe it's acceptable ?
